chia/cmds/cmd_classes.py

```diff
@@ -0,0 +1,322 @@
+from __future__ import annotations
+
+import asyncio
+import collections
+import inspect
+import sys
+from contextlib import asynccontextmanager
+from dataclasses import MISSING, dataclass, field, fields
+from typing import (
+    Any,
+    AsyncIterator,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Protocol,
+    Type,
+    Union,
+    get_args,
+    get_origin,
+    get_type_hints,
+)
+
+import click
+from typing_extensions import dataclass_transform
+
+from chia.cmds.cmds_util import get_wallet_client
+from chia.rpc.wallet_rpc_client import WalletRpcClient
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.byte_types import hexstr_to_bytes
+from chia.util.streamable import is_type_SpecificOptional
+
+SyncCmd = Callable[..., None]
+
+
+class SyncChiaCommand(Protocol):
+    def run(self) -> None: ...
+
+
+class AsyncChiaCommand(Protocol):
+    async def run(self) -> None: ...
+
+
+ChiaCommand = Union[SyncChiaCommand, AsyncChiaCommand]
+
+
+def option(*param_decls: str, **kwargs: Any) -> Any:
+    if sys.version_info < (3, 10):  # versions < 3.10 don't know about kw_only and they complain about lacks of defaults
+        # Can't get coverage on this because we only test on one version
+        default_default = None  # pragma: no cover
+    else:
+        default_default = MISSING
+
+    return field(  # pylint: disable=invalid-field-call
+        metadata=dict(
+            option_args=dict(
+                param_decls=tuple(param_decls),
+                **kwargs,
+            ),
+        ),
+        default=kwargs.get("default", default_default),
+    )
+
+
+class HexString(click.ParamType):
+    name = "hexstring"
+
+    def convert(self, value: str, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> bytes:
+        if isinstance(value, bytes):  # This if is due to some poor handling on click's part
+            return value
+        try:
+            return hexstr_to_bytes(value)
+        except ValueError:
+            self.fail(f"{value} is not a valid hex string", param, ctx)
+
+
+class HexString32(click.ParamType):
+    name = "hexstring32"
+
+    def convert(self, value: str, param: Optional[click.Parameter], ctx: Optional[click.Context]) -> bytes32:
+        if isinstance(value, bytes32):  # This if is due to some poor handling on click's part
+            return value
+        try:
+            return bytes32.from_hexstr(value)
+        except ValueError:
+            self.fail(f"{value} is not a valid 32-byte hex string", param, ctx)
+
+
+@dataclass(frozen=True)
+class _CommandParsingStage:
+    my_dataclass: Type[ChiaCommand]
+    my_option_decorators: List[Callable[[SyncCmd], SyncCmd]]
+    my_subclasses: Dict[str, _CommandParsingStage]
```

Comment:
These are the classes hinted for the attributes/parameters?  As opposed to classes that inherit from `_CommandParsingStage`.

Reply:
No, this hint is correct (if that's what your asking?). "subclasses" is an incorrect name though yes, more like "members": 4412a9935d

Reply:
That is where I was going with it.  `:]`  Just wanted to start with asking the question to make sure I had the right idea about what this is.